### ISSUE TITLE
Automatic closing of unintentional duplicate paths

### ISSUE THREAD

**Issue by tzaeschke** [quicwg/multipath#451]
```
[..]. To avoid this pitfall, endpoints could
adopt a simple coordination rule, such as only letting the client
initiate closure of duplicate paths, or perhaps relying on
the application protocol to decide which paths should be closed.
```

I understand this as: it is okay to just implement the simple rule,
and that letting the application protocol decide is an optional feature.

This should IMHO be more prescriptive and make sure that this also works for a path-aware network, where use of multiple paths for the same 4-tuple is very intentional. 

I think the application should always be able to decide on the policy and only use the simple rule if no application protocol is defined or if it indicates that the simple rule is fine.

I would propose to change the phrasing here to something like:

```
[..]. To avoid this pitfall, endpoints should allow the application
protocol to decide whether or which path to close. If the application
protocol defines no such rule, a simple default coordination rule 
could be adopted, such as only letting the client initiate closure of 
duplicate paths.
```

**Comment by mirjak** [comment]
I don't think this change makes a real difference. However, note that is it also possible to deploy the multipath extension without application awareness.

**Comment by tzaeschke** [comment]
I agree that application awareness should be optional from the point of view of application developers.
However, I think if a library supports application awareness at all, it definitely should support replacing the "simple rule" with something else.

My concern is that implementers may choose to hard code the "simple coordination rule", This would be problematic if an application with intentional duplicate path usage creates two paths with distinct 4-tuples and these two paths are then migrated to the same 4-tuple (the same example given in the text). If the "simple rule" is hard coded, this would cause one of the paths top disappear.

In other words: if an implementation supports duplicate paths _and_ supports application awareness, then it definitely _should_ support replacing the "simple rule" with something else.
One alternative is that the implementation is somehow aware that duplicate paths are desirable and in that case employs an alternative coordination rule of "don´t do anything, having duplicate paths is fine".

Does that make sense?

**Comment by huitema** [comment]
I am not sure that we should get into discussions of APIs. We do not specify an API in the draft, and I don't think that we would have enough experience to do that well. For example, it is not clear that stacks will let applications manage paths *at all* -- they could have simple policy rules to handle the path creation under the hood. I don't think we should add any text there.

**Comment by tzaeschke** [comment]
> I am not sure that we should get into discussions of APIs. 

I'm just a bit concerned that `or perhaps relying on the application protocol to decide which paths should be closed.` is understood as exactly that, an API suggestion. The suggestion being that even in presence of an API (application protocol), asking the application for a policy on on duplicate path closure is optional and can be dismissed.

I guess my concern is really with the words "the application protocol" (sounding like there is an API) vs "perhaps" (sounding almost dismissive to me).

How about: 

```
[..]. To avoid this pitfall, endpoints could
adopt a simple coordination rule, such as only letting the client
initiate closure of duplicate paths, or, if available, let the
the application protocol to decide which paths should be closed.
```
That would actually be shorter (by one character) than the current wording.

**Comment by mirjak** [comment]
To be honest, I don't really see the difference. You don't need a specific interface, because an application can already close a path anytime. So this is more saying, you may implement a default policy in the quic stack or just do nothing in the quic stack and rely on the application to the right thing (which might also be nothing).

**Comment by tzaeschke** [comment]
@mirjak Thanks, and my apologies, it turns out I managed to misread (even while rephrasing it) what it actually says here. It does say exactly what I wanted.
